src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -454,18 +471,19 @@ def _apply_matrix_matrix(node):
         assign = Assignment.create(result_ref.copy(), rhs)
         # Create ii loop and add the above code as a child
         # Work out the bounds
-        lower_bound, upper_bound, step = _get_array_bound(matrix1, 0)
+        lower_bound, upper_bound, step = _get_array_bound(matrix1, 1)
```

Comment:
Oops! Embarrassing :-( Are these changes needed to make LFRic give the same answers? (i.e. LFRic was broken and this fixes it?)

Reply:
Yes, without this (and preventing the substitution of cases with same lhs and rhs operand) LFric does not converge in some sub-stepping and fails after reaching a iteration threshold.